src/System Application/Test/AppSource Gallery/src/AppSourceProductManagerTest.Codeunit.al

```diff
@@ -0,0 +1,470 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.Apps.AppSource.Test;
+
+using System.Apps.AppSource;
+using System.RestClient;
+using System.Environment.Configuration;
+using System.TestLibraries.Utilities;
+
+codeunit 135074 "AppSource Product Manager Test" implements "IAppSource Product Manager Dependencies"
```

Comment:
From what I can see these tests calls out to the actual product catalogue on the internet. We should mock those calls, otherwise snap will start failing if there are any connection issues (say they start throttling us or anything similar).